src/Microsoft.DotNet.VersionTools/Automation/NupkgNameInfo.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.VersionTools.Automation
+{
+    class NupkgNameInfo
+    {
+        private static Regex s_nugetFileRegex = new Regex(
+            @"^(?<id>.*?)\.(?<version>([0-9]+\.)?[0-9]+\.[0-9]+(-(?<prerelease>[A-z0-9-]+))?)(?<symbols>\.symbols)?\.nupkg$");
+
+        public NupkgNameInfo(string path)
+        {
+            Match match = s_nugetFileRegex.Match(Path.GetFileName(path));
```

Comment:
Nice, I'll change it.

---
In reply to: [71019284](https://github.com/dotnet/buildtools/pull/863#discussion_r71019284) [](ancestors = 71019284)